File: Advanced Business With .NET/sprint-1/src/Domain/Entities/Dentista.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace UserApi.Domain
{
    public class Dentista
    {
        [BsonId]
        public ObjectId Id { get; set; }  // ID do banco
        public string? IdDentista { get; set; } = null!; 
        public string Nome { get; set; } = null!;
        public string Sobrenome { get; set; } = null!;
        public string Telefone { get; set; } = null!;
        public string IdClinica { get; set; } = null!;
        public string Idespecialidade { get; set; } = null!;
        public string Avaliacao { get; set; } = null!;
}
}

File: JAVA ADVANCED/sprint-2/challenge/src/main/java/br/com/fiap/challenge/domains/Dentista.java
package br.com.fiap.challenge.domains;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Table(name = "dentista")
public class Dentista {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id_dentista", nullable = false)
    private String idDentista;

    @Column(name = "nome", length = 100, nullable = false)
    private String nome;

    @Column(name = "sobrenome", length = 100, nullable = false)
    private String sobrenome;

    @Column(name = "telefone", length = 15, nullable = false)
    private String telefone;

    @ManyToOne
    @JoinColumn(name = "id_clinica", nullable = false)
    private Clinica clinica;

    @ManyToOne
    @JoinColumn(name = "id_especialidade", nullable = false)
    private Especialidade especialidade;

    @Column(name = "avaliacao", nullable = false)
    private Float avaliacao;

}
